File: PlanIT-backend/src/main/java/com/example/demo/model/Room.java
package com.example.demo.model;

import javax.persistence.*;
import java.util.ArrayList;
import java.util.List;

@Entity
@Table(name = "rooms")
public class Room {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    private Integer id;

    private String roomName;

    @ManyToMany
    @JoinTable(name = "room_user",
            joinColumns = {@JoinColumn(name = "room_id")},
            inverseJoinColumns = {@JoinColumn(name = "user_id")})
    List<User> users;

    @OneToMany(fetch = FetchType.EAGER, mappedBy = "room")
    List<Message> messages;

    public Room(String room_name) {
        this.roomName = room_name;
        users = new ArrayList<>();
    }

    public Room() {
    }

    public List<Message> getMessages() {
        return messages;
    }

    public void setMessages(List<Message> messages) {
        this.messages = messages;
    }

    public Integer getId() {
        return id;
    }

    public void setId(Integer id) {
        this.id = id;
    }

    public String getRoomName() {
        return roomName;
    }

    public void setRoomName(String roomName) {
        this.roomName = roomName;
    }

    public List<User> getUsers() {
        return users;
    }

    public void setUsers(List<User> users) {
        this.users = users;
    }

    @Override
    public String toString() {
        return "Room{" +
                "id=" + id +
                ", room_name='" + roomName + '\'' +
                '}';
    }
}

File: PlanIT-frontend/src/app/model/Room.ts
import {User} from "./User";
import {Message} from "./Message";

export class Room{
  private _roomName: string;
  private _users: User[];
  private _messages: Message[];


  constructor(roomName: string, users: User[], messages: Message[]) {
    this._roomName = roomName;
    this._users = users;
    this._messages = messages;
  }


  get roomName(): string {
    return this._roomName;
  }

  set roomName(value: string) {
    this._roomName = value;
  }

  get users(): User[] {
    return this._users;
  }

  set users(value: User[]) {
    this._users = value;
  }

  get messages(): Message[] {
    return this._messages;
  }

  set messages(value: Message[]) {
    this._messages = value;
  }
}
